tests/cbmc/proofs/s2n_stuffer_raw_write/s2n_stuffer_raw_write_harness.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <sys/param.h>
+
+#include "api/s2n.h"
+#include "stuffer/s2n_stuffer.h"
+
+#include <assert.h>
+#include <cbmc_proof/cbmc_utils.h>
+#include <cbmc_proof/make_common_datastructures.h>
+#include <cbmc_proof/proof_allocators.h>
+
+void s2n_stuffer_raw_write_harness() {
+    /* Non-deterministic inputs. */
+    struct s2n_stuffer *stuffer = cbmc_allocate_s2n_stuffer();
+    __CPROVER_assume(s2n_stuffer_is_valid(stuffer));
+    uint32_t data_len;
+
+    /* Non-deterministically set initialized (in s2n_mem) to true. */
```

Comment:
This seems like it needs to happen for every proof, and is easy to forget.  WE should have a mechanism for it.

Reply:
We actually have a mechanism for that, the `--nodet-static` flag in CBMC. However, it led to spurious error in static function calls. I create this branch https://github.com/feliperodri/s2n/tree/diffblue-nondet-static-error to reproduce the error. I also created the issue https://github.com/awslabs/s2n/issues/2010 so we can clean up all proof harnesses using it  on a separate PR.